xds/internal/client/xds.go

```diff
@@ -170,9 +173,23 @@ func processHTTPFilterOverrides(cfgs map[string]*anypb.Any) (map[string]httpfilt
 	}
 	m := make(map[string]httpfilter.FilterConfig)
 	for name, cfg := range cfgs {
-		_, config, err := validateHTTPFilterConfig(cfg, false)
+		optional := false
+		s := new(v3routepb.FilterConfig)
+		if ptypes.Is(cfg, s) {
+			if err := ptypes.UnmarshalAny(cfg, s); err != nil {
+				return nil, fmt.Errorf("filter override %q: error unmarshalling FilterConfig: %v", name, err)
+			}
+			cfg = s.GetConfig()
+			optional = s.GetIsOptional()
+		}
+
+		httpFilter, config, err := validateHTTPFilterConfig(cfg, false, optional)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("filter override %q: %v", name, err)
+		}
+		if httpFilter == nil {
```

Comment:
Is it slightly better to check `config` instead?
It feels natural(?)

What if `httpFilter` is non-nil, but `cfg` is nil?
We might want to store the nil cfg in map? In that case, checking `httpFilter` is the right thing to do.